### DISCUSSION TITLE
Saving data with php backend

### DISCUSSION THREAD

**Discussion by Swapnilchavan18** [codex-team/editor.js#1895] [Q&A]
Hi, I have searched over google, but don't find the correct way to save editorjs output, when we try to save with json there are lot of issues with slashes, JSON decode will not work and will basically remove content with double quotes(after using stripslashes and html_entity_decode) like href, any classes, attributes, etc

What is the best way to save the data that we can also process at PHP backend. Thanks in advance.

```
{
            "id" : "xEhA7tB3HH",
            "type" : "paragraph",
            "data" : {
                "text" : "There are dozens of <a href=\"https://github.com/editor-js\">ready-to-use Blocks</a> and the <a href=\"https://editorjs.io/creating-a-block-tool\">simple API</a> for creation any Block you need. For example, you can implement Blocks for Tweets, Instagram posts, surveys and polls, CTA-buttons and even games."
            }
        }
```

**Comment by kaaaaaaaaaaai** [comment]
@Swapnilchavan18 is it solved?

**Comment by victor-david** [comment]
### Sending
To send data to the backend, I base64 encode the json 

~~~javascript
const contentToSend = encode64(JSON.stringify(data));  // data is the editor output
~~~

At the PHP end, I base 64 decode to get the JSON again, and then save it to the database.
~~~php
$contentForDb = base64_decode($contentSent ?: '');
// save to db
~~~

### Retreiving
Later, in PHP, to restore, I get the json from the db, base64 encode it, and send it to the browser. where Javascript base64 decodes it, and then recreates the data object to init the editor

~~~javascript
const dataForEditor = JSON.parse(decode64(data64)); // data64 came from PHP
~~~

-  The **encode64()** function (not shown) uses native  btoa() to handle unicode
- The **decode64()** function(not shown) uses native atob() to handle unicode

Hopefully, that helps a little. I've had quite a bit of trial and error.
